File: CastIt/Server/Dtos/Responses/FileItemResponseDto.cs
namespace CastIt.Server.Dtos.Responses
{
    public class FileItemResponseDto
    {
        public long Id { get; set; }
        public int Position { get; set; }
        public string Path { get; set; }
        public double PlayedSeconds { get; set; }
        public double TotalSeconds { get; set; }
        public double PlayedPercentage { get; set; }
        public bool IsBeingPlayed { get; set; }
        public long PlayListId { get; set; }
        public bool IsLocalFile { get; set; }
        public bool IsUrlFile { get; set; }
        public bool Exists { get; set; }
        public string Filename { get; set; }
        public string Size { get; set; }
        public string Extension { get; set; }
        public bool Loop { get; set; }
        public string SubTitle { get; set; }
    }
}

File: CastIt.Client/ios/CastIt Watch App/Models/Responses/FileItemResponseDto.swift
import Foundation

// Mirrors ClientApp IFileItemResponseDto
struct FileItemResponseDto: Codable, Identifiable {
    let id: Int
    let name: String
    let description: String
    let totalSeconds: Double
    let path: String
    let position: Int
    let playedPercentage: Double
    let playListId: Int
    let loop: Bool
    let lastPlayedDate: String?

    let isBeingPlayed: Bool
    //let type: AppFile
    let isLocalFile: Bool
    let isUrlFile: Bool
    let playedSeconds: Double
    let canStartPlayingFromCurrentPercentage: Bool
    let wasPlayed: Bool
    let isCached: Bool

    let exists: Bool
    let filename: String
    let size: String
    let `extension`: String

    let subTitle: String
    let resolution: String?
    let duration: String
    let playedTime: String
    let totalDuration: String
    let fullTotalDuration: String
    let thumbnailUrl: String?

    let currentFileVideos: [FileItemOptionsResponseDto]
    let currentFileAudios: [FileItemOptionsResponseDto]
    let currentFileSubTitles: [FileItemOptionsResponseDto]
    let currentFileQualities: [FileItemOptionsResponseDto]
    let currentFileVideoStreamIndex: Int
    let currentFileAudioStreamIndex: Int
    let currentFileSubTitleStreamIndex: Int
    let currentFileQuality: Int
}
